telemetry/src/lib.rs

```diff
@@ -269,3 +288,46 @@ impl TelemetryFlusher {
         }
     }
 }
+
+/// Shared struct to add to any struct that needs to be traced over dbus.
+#[cfg(feature = "zbus-tracing")]
+#[derive(
+    Debug,
+    Default,
+    Clone,
+    Eq,
+    PartialEq,
+    zbus::zvariant::Type,
+    zbus::zvariant::SerializeDict,
+    zbus::zvariant::DeserializeDict,
+)]
+pub struct Tracing {
+    pub ctx: HashMap<String, String>,
+}
+
+#[cfg(feature = "zbus-tracing")]
+impl Tracing {
+    pub fn none() -> Self {
+        Self {
+            ctx: HashMap::new(),
+        }
+    }
+
+    #[cfg(feature = "otel")]
```

Comment:
I didn't fully understand this code. Can you add some docs including an [example](https://doc.rust-lang.org/rustdoc/write-documentation/documentation-tests.html) of how to use?